### ISSUE TITLE
Map size being set to zero

### ISSUE THREAD

**Issue by ehovekamp** [KSneijders/AoE2ScenarioParser#43]
If one attempts to set map size to the value it already holds, the map size is set to 0. 
```
map_manager.map_size = 10
print(map_manager.map_size)
map_manager.map_size = 10
print(map_manager.map_size)
```

Gives
```
10
0
```
Easy enough to work around with an If statement to check if the value is already what you want, but this seems like unintended behavior.

**Comment by KSneijders** [comment]
Hey, thanks for reporting! That's indeed not intended behaviour at all 11

Just to make sure, could you tell me the version of the parser you're using? If it's not the latest (0.1.57) could you try and upgrade and see if the issue persists? 

Thanks!

**Comment by ehovekamp** [comment]
Checked my version, it's 0.1.57

Maybe I can save you some time, I've looked through the code and I think I see the issue. 
From map_manager.py:
```
 def map_size(self, new_size: int):
        old_size = self._map_width
        difference = new_size - old_size

        self._map_width = new_size
        self._map_height = new_size

        new_terrain = []
        if difference < 0:
            # Remove ends of rows (x) & remove final rows entirely (y)
            for index, chunk in enumerate(list_chuncks(self.terrain, old_size)):
                if index == new_size:
                    break
                new_terrain.extend(chunk[:new_size])
        elif difference > 0:
            # Add ends to rows (x) & add entirely new rows  (y)
            chunk_gen = list_chuncks(self.terrain, old_size)
            for index in range(new_size):
                if index < old_size:
                    row = next(chunk_gen) + [TerrainTile(uuid=self._uuid) for _ in range(difference)]
                else:
                    row = [TerrainTile(uuid=self._uuid) for _ in range(new_size)]
                new_terrain.extend(row)
        self.terrain = new_terrain
```

Its sets new_terrain as empty, and then only puts something into new_terrain if difference is greater or less than 0. So if difference is equal to 0 then it just sets self.terrain to empty.

**Comment by KSneijders** [comment]
@ehovekamp Thank you very much for checking the code! That's quite an oversight 11
I was unable to take a look at the issue earlier (work) so this helps! ❤️

**Comment by KSneijders** [comment]
@ehovekamp A fix is now live on version `0.1.58`.  
You can update the parser using the following command:

```
pip install --no-cache-dir --upgrade AoE2ScenarioParser
```
